### PR TITLE
Split link-at-beginning watch into new reason

### PR DESCRIPTION
**Pull request by NobodyNada**
[Discussed in CHQ](https://chat.stackexchange.com/transcript/11540?m=54586828#54586828). The reason has its own whitelist for now, until we have unified whitelist-checking.

### CODE REVIEW

**Comment by user12986714** [comment]
If we are implementing this as a separate rule (which I think we should), I would just duplicate the code used for link-at-end, which takes whitelists into considerations.
```python
@create_rule("link at end of {}", title=False, all=False, sites=[
    "superuser.com", "askubuntu.com", "drupal.stackexchange.com", "meta.stackexchange.com",
    "security.stackexchange.com", "patents.stackexchange.com", "money.stackexchange.com",
    "gaming.stackexchange.com", "arduino.stackexchange.com", "workplace.stackexchange.com"])
def link_at_end(s, site):   # link at end of question, on selected sites
    s = regex.sub("</?(?:strong|em|p)>", "", s)
    match = regex.compile(r"(?i)https?://(?:[.A-Za-z0-9-]*/?[.A-Za-z0-9-]*/?|plus\.google\.com/"
                          r"[\w/]*|www\.pinterest\.com/pin/[\d/]*)(?=</a>\s*$)").search(s)
    if match and not is_whitelisted_website(match.group(0)):
        return True, u"Link at end: {}".format(match.group(0))
    return False, ""
```

**Comment by NobodyNada** [comment]
See the CHQ discussion; my understanding is that this reason will eventually adopt the MS whitelist once we have the new whitelist-checker.

**Comment by makyen** [comment]
This should have a `max_rep` and a `max_score`. The only data we have is for what the watchlist is using, which is `max_rep=32, max_score=1`.

File: findspam.py [hunk 1]
```diff
@@ -2265,6 +2265,14 @@ def format_with_city_list(regex_text):
             all=False, sites=["workplace.stackexchange.com", "workplace.meta.stackexchange.com"],
             username=True, body_summary=False, body=False, title=False,
             max_rep=93, max_score=1)
-
+# Link at beginning of post; pulled from watchlist
+create_rule("link at beginning of {}",
+            r'^\s*<p>\s*(?:</?\w+/?>\s*)*<a href="(?!(?:[a-z]+:)?//(?:[^" >/.]*\.)*(?:(?:quora|medium'
+            r'|googleusercontent|youtube|microsoft|unity3d|wso|merriam-webster|oracle|magento|example'
+            r'l|apple|google|github|imgur|stackexchange|stackoverflow|serverfault|superuser|askubuntu)\.com'
```
Comment (makyen) [inline findspam.py:2272]: There's a minor editing error in this line. It should start with `|`, not `l`. The first letter `l` is extra.

Comment (NobodyNada) [reply findspam.py]: Oops, good catch!

**Comment by NobodyNada** [comment]
@makyen Thanks. I went ahead and put in a max_rep and max_score for now; I've noticed enough false-positives from this that I think it's worthwhile.

**Comment by makyen** [comment]
I agree, I'd rather have the `max_rep` and `max_score`, at least for now. Maybe once we have the whitelist working, it will reduce the FP rate sufficiently such that we can look at adjusting those, if it's felt there's a need.


!!/approve

**Pull request merged**